wire-library/wire-runtime-swift/src/main/swift/AnyMessage.swift

```diff
@@ -0,0 +1,118 @@
+/*
+ * Copyright 2022 Block Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import Foundation
+
+/**
+ * Wire implementation of the `google.protobuf.Any` type. The `Any` type wraps an arbitrary
+ * protobuf message. Field of `Any` generated by Wire will be typed [AnyMessage]. [typeUrl] is
+ * used to identify the type of the message, defined as 'type.googleapis.com/full.type.name'.
+ * [value] is the serialized representation of the wrapped message. The [pack] and [unpack] methods
+ * are available to wrap and unwrap an `Any` message.
+ *
+ * Example
+ * ```
+ * let anyMessage = try AnyMessage.pack(person)
+ * let person = try anyMessage.unpack(Person.self)
+ * ```
+ */
+public struct AnyMessage {
+    public let typeUrl: String
+    public let value: Data
+
+    public init(typeUrl: String, value: Data) {
+        self.typeUrl = typeUrl
+        self.value = value
+    }
+
+    public enum DecodingError: Swift.Error, Equatable {
+        case typeUrlMismatch
+    }
+
+    public func unpack<Output: ProtoMessage & ProtoDecodable>(
+        _ type: Output.Type,
+        enumDecodingStrategy: ProtoDecoder.UnknownEnumValueDecodingStrategy = .returnNil
```

Comment:
The default strategy on `ProtoDecoder` is `.throwError`, so at the least we should match that for consistency. But perhaps as a better alternative, we should pass in the full decoder (with a default value). It allows for reuse of the same decoder if desired, and also means that if we have other customizations available on the decoder in the future, then this code won't need to change.

As an optimization, we can make the default a statically allocated, reused value, like `static let defaultDecoder: Decoder = .init()`.

Reply:
Sounds good regarding passing a full `ProtoDecoder` - updated.

Regarding having a `static` default, it feels a bit weird to have that live on `AnyMessage` since decoders are used for all messages. Maybe we make that universal as a separate change if that's something we want to do?